pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -181,7 +183,7 @@ func ensure(flowcontrolClientSet flowcontrolclient.FlowcontrolV1alpha1Interface,
 	return nil
 }
 
-func upgrade(flowcontrolClientSet flowcontrolclient.FlowcontrolV1alpha1Interface, flowSchemas []*flowcontrolv1alpha1.FlowSchema, priorityLevels []*flowcontrolv1alpha1.PriorityLevelConfiguration) error {
+func upgrade(flowcontrolClientSet flowcontrolclient.FlowcontrolV1beta1Interface, flowSchemas []*flowcontrolv1beta1.FlowSchema, priorityLevels []*flowcontrolv1beta1.PriorityLevelConfiguration) error {
```

Comment:
this isn't new in this PR, but what is this method doing? it looks like it is reading existing objects via v1alpha1, setting defaults, then comparing to see if there was any change... given defaults are applied to the object when it is read via the Get, if there is any difference, we have a bug in the API server or P&F defaulting path

Reply:
The point of this function is not defaulting, it is to ensure that these objects have exactly the intended spec.
An explanatory comment would be a good add here, and also on `ensure`; I find both names too short and vague to make their meaning clear.

Reply:
oh, I see... this is stomping API-persisted objects with the specs for the bootstrap schemas, so if an n-1 API server starts last, it will persist its view of what the bootstrap schemas should look like

Reply:
I don't fully understand these, so I hope someone else more knowledgable could add a comment in a follow-up PR.

Reply:
Additional context:
* we make the objects so humans know what the system is doing, they don't actually need to be present for the request path to function, it will assume they are there if it doesn't see them
* apiservers which disagree about the definition of these will fight. We made them fight slowly.